stacks-common/src/util/vrf.rs

```diff
@@ -290,12 +296,12 @@ impl VRFProof {
         let c_bytes = c.reduce().to_bytes();
 
         // upper 16 bytes of c must be 0's
-        for i in 16..32 {
-            if c_bytes[i] != 0 {
```

Comment:
I think this loses semantic information.  The algorithm specifically calls for checking the upper 16 bytes of  `c`, and the original `for`-loop makes this clearer.

Reply:
Certainly not a hill I will die on, but I think its best to avoid index accesses if possible and generally find it easier to reason about bytes in an array using the "byte in bytes.iter()" syntax. However, this is not my codebase and I will revert as desired :D

Reply:
I agree about avoiding looping with an index access in general, but I also think there's a way to get more semantic information from the iteration:

```rust
        for c_byte in c_bytes[16..32].iter() {
            if *c_byte != 0 {
                return false;
            }
        }
```

Reply:
Thanks :pray: